server/zio-http/src/main/scala/sttp/tapir/server/zhttp/ZHttpInterpreter.scala

```diff
@@ -0,0 +1,140 @@
+package sttp.tapir.server.zhttp
+
+import io.netty.handler.codec.http.HttpResponseStatus
+import sttp.capabilities
+import sttp.capabilities.zio.ZioStreams
+import sttp.model.{HasHeaders, QueryParams, Uri, Header => SttpHeader, Method => SttpMethod}
+import sttp.monad.MonadError
+import sttp.tapir.model.{ConnectionInfo, ServerRequest}
+import sttp.tapir.server.interpreter.{RequestBody, ServerInterpreter, ToResponseBody}
+import sttp.tapir.{CodecFormat, Endpoint, RawBodyType, RawPart, WebSocketBodyOutput}
+import zhttp.http.{CanSupportPartial, Http, HttpChannel, HttpContent, Request, Response, Status, Header => ZHttpHeader}
+import zio._
+import zio.stream._
+
+import java.net.InetSocketAddress
+import java.nio.charset.{Charset, StandardCharsets}
+import scala.collection.immutable.Seq
+
+class ZHttpServerRequest(req: Request) extends ServerRequest {
+  def protocol: String = "HTTP/1.1" //TODO
+
+  def local =
+    for {
+      host <- req.url.host
+      port <- req.url.port
+    } yield new InetSocketAddress(host, port)
+
+  lazy val connectionInfo: ConnectionInfo =
+    ConnectionInfo(local, None, None)
+
+  def underlying: Any = req
+
+  lazy val pathSegments: List[String] = req.url.path.toList
+  lazy val queryParameters: QueryParams = QueryParams.fromMap(Map.empty) // TODO: parse
+
+  def method: SttpMethod = SttpMethod(req.method.asJHttpMethod.name().toUpperCase)
+
+  def uri: Uri = Uri.unsafeParse(req.url.toString()) //TODO: is this correct?
```

Comment:
yes looks correct - you can then implement `queryParameters` in terms of `uri`, as it has the `QueryParams` instance inside

Reply:
Can it fail (as the name implies) ?What would happen if the parsing would fail?

Reply:
Yes, an exception will be thrown if the uri is invalid. But since it reached the server, it should be safe to assume that it is valid :)

Reply:
Great!